service/pipeline.ml

```diff
@@ -335,8 +335,9 @@ end
 let latch ~label x =
   let prev = ref (Error (`Active `Ready), None) in
   Current.component "latch %s" label |>
-  let> x = Current.state ~hidden:true x in
-  Result.iter (fun x -> prev := Ok x, None) x;
+  let> x = Current.state ~hidden:true x
+  and> meta = Current.Analysis.metadata x in
+  Result.iter (fun x -> prev := Ok x, meta) x;
```

Comment:
There seem to be some issues here. The metadata and the state aren't in sync which can cause problem:
![scrn-2023-01-30-19-58-47](https://user-images.githubusercontent.com/2611789/215582557-642b51da-5e5c-4d8d-a9e8-cd9c2b211139.png)
Here is the log (and previous working logs): https://opam-repo-ci.ci3.ocamllabs.io/job/2023-01-30/183948-opam-ci-analyse-9093d6